### PR TITLE
WP Super Cache Landing Page and Settings restructure

### PR DESCRIPTION
**Pull request by mikestottuk**
#### Changes proposed in this Pull Request
This PR restructures the plugins settings and splits them into a top level menu and breaks out each tab of the old settings page into sub menu pages. Doing this allowed for the opportunity to introduce a new landing page for WP Super Cache with streamlined install and activation options for Jetpack Boost.

#### Other information:
- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion
This is covered here `pc9hqz-1e4`

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
I've tested the plugin throughout and it all looks to be working OK. The bulk of the work just impacts the settings pages not the actual plugin code itself. On a fresh install, without Jetpack Boost installed or active:-

- [ ] Install and activate WP Super Cache
- [ ] Visit the new homepage
- [ ] Enable caching
- [ ] Go to Advanced settings from the home page - enable caching for logged in users.
- [ ] Visit a few pages on your site so they are cached
- [ ] Check the cached pages box updates
- [ ] On the new homepage, click the button to Install boost - make sure this installs and activates OK
- [ ] On plugins.php, deactivate Jetpack Boost
- [ ] Go to the "Easy tab" it should just show a message to Activate Boost (in the other banner) check this works
- [ ] On plugins.php deactivate and **delete** Jetpack Boost
- [ ] Go to the Advanced tab
- [ ] On plugins.php deactivate and **delete** Jetpack Boost (again)
- [ ] Dismiss the install Boost prompt from the homepage - check it stays gone
- [ ] Check its still gone on the settings pages
- [ ] Clear the option in the database
- [ ] On plugins.php deactivate and **delete** Jetpack Boost (again)
- [ ] Dismiss the install Boost prompt from any of the settings page - check it stays gone
- [ ] Check it also stays gone on the home page
- [ ] On plugins.php deactivate and **delete** Jetpack Boost (again, last time I promise)
- [ ]  Clear the option in the database (again, final time)
- [ ] Go to "Add New" plugin and search for Jetpack Boost - install it but **do not activate**
- [ ] Go back to the homepage it should show "Activate Boost", same on the settings page.
- [ ] Check the "Cached files" tab. List all files
- [ ] Clear the cache from this page
- [ ] Visit the Easy page, delete the cache
- [ ] Visit Advanced, restore factory settings (at the bottom)
- [ ] Check that checking the settings works OK and save down
- [ ] Generate a lot of dummy content, you can use [Theme Test data for this](https://github.com/WPTT/theme-test-data)
- [ ] Preload the cache, check it is working (takes a bit of time to cache everything).
- [ ] The cached pages formats the number as 1.0k, etc depending on how many pages (might need to generate a load of dummy posts for that!)
- [ ] It will also do 1.0m and 1.0b - don't generate this many, if you want to test just hard code the count in `function wpsc_cached_pages(){..}` see screenshots below
- [ ] Enable debugging, check it is still outputting
- [ ] Test the homepage layout on mobile
- [ ] Test the homepage layout on tablet
- [ ] Check the plugins tab and enable a plugin
- [ ] Check the CDN tab, enable Jetpack (site accelerator off) check it again, turn on site accelerator (check it again)

All this should have coverage of all the settings pages and that they're working as intended.

### CODE REVIEW

**Comment by oskosk** [comment]
This is excellent work!

**Comment by oskosk** [comment]
I'm seeing a `Notice: Undefined variable: language in /srv/users/userb144921b/apps/userb144921b/public/wp-content/plugins/wp-super-cache-dev/wp-cache.php on line 308` notice while visiting the new page (`wp-admin/admin.php?page=wp-super-cache-home`)

**Comment by mikestottuk** [comment]
> I'm seeing a `Notice: Undefined variable: language in /srv/users/userb144921b/apps/userb144921b/public/wp-content/plugins/wp-super-cache-dev/wp-cache.php on line 308` notice while visiting the new page (`wp-admin/admin.php?page=wp-super-cache-home`)

Thanks, still testing things and trying to get it past the tests.

File: projects/plugins/super-cache/wp-cache.php [hunk 1]
```diff
@@ -3855,6 +4483,5 @@ function wpsc_render_partial( $partial, array $page_vars = array() ) {
 	foreach ( $page_vars as $key => $val ) {
 		$$key = $val;
 	}
-	global $current_user;
```
Comment (mikestottuk) [inline projects/plugins/super-cache/wp-cache.php:3868]: this isn't used in the function

**Comment by thingalon** [comment]
I have merged this into `add/boost-banner` and will finish it there.

**Comment by thingalon** [comment]
I have revived the branch but not the PR because this code still needs to be build on & integrated, but through a separate PR.

**Pull request merged**